server/src/main/java/org/opensearch/node/remotestore/RemoteStoreNodeService.java

```diff
@@ -0,0 +1,156 @@
+        Setting.Property.NodeScope
+    );
+
+    /**
+     * Node join compatibility mode introduced with remote backed storage.
+     *
+     * @opensearch.internal
+     */
+    public enum CompatibilityMode {
+        STRICT("strict");
+
+        public final String mode;
+
+        CompatibilityMode(String mode) {
+            this.mode = mode;
+        }
+
+        public static CompatibilityMode parseString(String compatibilityMode) {
+            try {
+                return CompatibilityMode.valueOf(compatibilityMode.toUpperCase(Locale.ROOT));
+            } catch (IllegalArgumentException e) {
+                throw new IllegalArgumentException(
+                    "["
+                        + compatibilityMode
+                        + "] compatibility mode is not supported. "
+                        + "supported modes are ["
+                        + CompatibilityMode.values().toString()
+                        + "]"
+                );
+            }
+        }
+    }
+
+    public RemoteStoreNodeService(Supplier<RepositoriesService> repositoriesService, ThreadPool threadPool) {
+        this.repositoriesService = repositoriesService;
+        this.threadPool = threadPool;
+    }
+
+    /**
+     * Creates a repository during a node startup and performs verification by invoking verify method against
+     * mentioned repository. This verification will happen on a local node to validate if the node is able to connect
+     * to the repository with appropriate permissions.
+     * If the creation or verification fails this will close all the repositories this method created and throw
+     * exception.
+     */
+    public List<Repository> createAndVerifyRepositories(DiscoveryNode localNode) {
+        RemoteStoreNodeAttribute nodeAttribute = new RemoteStoreNodeAttribute(localNode);
+        RepositoriesService reposService = repositoriesService.get();
+        List<Repository> repositories = new ArrayList<>();
+        for (RepositoryMetadata repositoryMetadata : nodeAttribute.getRepositoriesMetadata().repositories()) {
+            String repositoryName = repositoryMetadata.name();
+            Repository repository;
+            RepositoriesService.validate(repositoryName);
+
+            // Create Repository
+            repository = reposService.createRepository(repositoryMetadata);
+            logger.info(
+                "remote backed storage repository with name [{}] and type [{}] created",
+                repository.getMetadata().name(),
+                repository.getMetadata().type()
+            );
+
+            // Verify Repository
+            String verificationToken = repository.startVerification();
+            repository.verify(verificationToken, localNode);
+            repository.endVerification(verificationToken);
+            logger.info(() -> new ParameterizedMessage("successfully verified [{}] repository", repositoryName));
+            repositories.add(repository);
```

Comment:
How are these repositories being used? when the cluster state is applied, these repos will be created again. The repos created here are not used later.

Reply:
These should be initialized in RepositoriesService

Reply:
I think we should retain the create repository and ensure repository services map is populated. This will help us recover from quorum loss scenarios where the cluster state isn't available.
We should then add a check to skip creating repository if it already exists